### PR TITLE
Infer prim tags from equivalent aten ones

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129686
* __->__ #130367

Take intersection of all the tags for corresponding aten op overloads. Previously, some of the rng ops not having tags caused issues with constant folding (they should get decomposed but thats a separate issue).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -329,6 +329,19 @@ def _backend_select_impl(*args, **kwargs):
     _prim = _prim_packet.default
     if tags:
         _prim._tags = tags
+    elif aten_packet := getattr(torch.ops.aten, name, None):
+        overload_tags = [
+            getattr(aten_packet, overload).tags for overload in aten_packet.overloads()
+        ]
+        tags_intersection = set(overload_tags[0])
```
Comment (ezyang) [inline torch/_prims/__init__.py:336]: Can't you just use your new OrderedSet instead

**Review by ezyang — APPROVED** [review]
I guess this is OK.... you need to fix tests though

**Comment by eellison** [comment]
The tests were incorrect prior. It's fixing them (although not lowering to inductor - tbd)

File: torch/_prims/__init__.py (hunk 1 above)
Comment (eellison) [reply torch/_prims/__init__.py:336]: not landed yet.. can do in follow ups

**Comment by eellison** [comment]
@pytorhbot merge

**Pull request closed without merging**